akka-stream/src/main/scala/akka/stream/javadsl/SourceWithContext.scala

```diff
@@ -220,6 +220,48 @@ final class SourceWithContext[+Out, +Ctx, +Mat](delegate: scaladsl.SourceWithCon
   def log(name: String): SourceWithContext[Out, Ctx, Mat] =
     this.log(name, ConstantFun.javaIdentityFunction[Out], null)
 
+  /**
+   * Context-preserving variant of [[akka.stream.javadsl.Source.logWithMarker]].
+   *
+   * @see [[akka.stream.javadsl.Source.logWithMarker]]
+   */
+  def logWithMarker(
+      name: String,
+      marker: function.Function[Out, LogMarker],
```

Comment:
Since markers are used for logging, I think we should pass the context to the marker function as well.

I did something like this when I was implementing it for my work.

```
implicit final class SourceWithContextOps[A, Ctx, Mat](val target: SourceWithContext[A, Ctx, Mat]) extends AnyVal {
    def logMarker(name: String, extract: (A, Ctx) => Any = (_, _), marker: (A, Ctx) => LogMarker)(implicit log: MarkerLoggingAdapter): SourceWithContext[A, Ctx, Mat] =
      target.asSource.logMarker(name, extract.tupled, marker.tupled).asSourceWithContext(_._2).map(_._1)
  }
```
Not suggesting to use implicit class, though but highlighting the `logMarker` method.